lib/routes/dhu/jiaowu/news.ts

```diff
@@ -26,32 +28,56 @@ export const route: Route = {
     name: '教务处通知',
     maintainers: ['KiraKiseki'],
     handler,
-    description: `| 学生专栏 | 教师专栏 |
-  | -------- | -------- |
-  | student  | teacher  |`,
+    description: `| 学生专栏 | 教师专栏 | 选课专栏（仅选课期间开放） | 辅修专业 |
+  | -------- | -------- | -------- | -------- |
+  | student  | teacher  | class    | fxzy     |`,
 };
 
 async function handler(ctx) {
-    const type = ctx.req.param('type');
-    const link = Object.hasOwn(map, type) ? `${base_url}${map[type]}` : `${base_url}/tzggwxszl/list.htm`;
-    const response = await got({
-        method: 'get',
-        url: link,
-        headers: {
-            Referer: base_url,
-        },
-    });
-
-    const $ = load(response.data);
+    const type = ctx.req.param('type') || 'student';
+    const link = `${baseUrl}${map[type]}`;
+    const { data: response } = await got(link);
+
+    const $ = load(response);
+
+    const items = await Promise.all(
+        $('.list2 > li')
+            .toArray()
+            .map(async (item) => {
+                item = $(item);
+                const newsTitle = item.find('.news_title > a');
+                const newsMeta = item.find('.news_meta');
+
+                // article meta
+                const link = newsTitle.attr('href');
+                const title = newsTitle.text();
+                const pubDate = parseDate(newsMeta.text(), 'YYYY-MM-DD', 'zh-cn');
+
+                // article content
+                const url = `${baseUrl}${link}`;
+                const description = await cache.tryGet(url, async () => {
+                    // some contents are only available for internal network
+                    try {
+                        const { data: response } = await got(url);
+                        const $ = load(response);
+                        return $('.wp_articlecontent').first().html() ?? '';
+                    } catch {
+                        return '';
+                    }
+                });
```

Comment:
Please cache the whole returned object
https://github.com/DIYgod/RSSHub/blob/eb82799227eb01f8cc255aba2387757b733e9cbf/lib/routes/dhu/jiaowu/news.ts#L69-L74
instead of `description` only.